Kerberos authentication fails to authorize client

##### Versions

v1.28.0

I am getting the exact same errors described in #1658, both client-side and broker-side

Client-side:
```
client/metadata got error from broker -1 while fetching metadata: EOF
```

Broker:
```
[2021-05-04 18:13:03,100] DEBUG [SocketServer brokerId=1] Connection with $IP/$IP disconnected (org.apache.kafka.common.network.Selector)
javax.security.sasl.SaslException: GSS initiate failed [Caused by GSSException: DerValue.getOID, not an OID 48]
        at jdk.security.jgss/com.sun.security.sasl.gsskerb.GssKrb5Server.evaluateResponse(GssKrb5Server.java:199)
        at org.apache.kafka.common.security.authenticator.SaslServerAuthenticator.handleSaslToken(SaslServerAuthenticator.java:419)
        at org.apache.kafka.common.security.authenticator.SaslServerAuthenticator.authenticate(SaslServerAuthenticator.java:294)
        at org.apache.kafka.common.network.KafkaChannel.prepare(KafkaChannel.java:185)
        at org.apache.kafka.common.network.Selector.pollSelectionKeys(Selector.java:550)
        at org.apache.kafka.common.network.Selector.poll(Selector.java:486)
        at kafka.network.Processor.poll(SocketServer.scala:904)
        at kafka.network.Processor.run(SocketServer.scala:803)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: GSSException: DerValue.getOID, not an OID 48
        at java.security.jgss/sun.security.jgss.GSSContextImpl.acceptSecContext(GSSContextImpl.java:381)
        at java.security.jgss/sun.security.jgss.GSSContextImpl.acceptSecContext(GSSContextImpl.java:303)
        at jdk.security.jgss/com.sun.security.sasl.gsskerb.GssKrb5Server.evaluateResponse(GssKrb5Server.java:167)
        ... 8 more
```

I have confirmed that v1.26.0 and v1.28.0 fail the same way, while v1.27.0 succeeds.

Seems like https://github.com/Shopify/sarama/commit/acced829c343f3b05ebae3ada1eca93feb14762c undid #1697, and that part should be reverted

--- a/go.mod
+++ b/go.mod
@@ -10,6 +10,7 @@ require (
 	github.com/eapache/queue v1.1.0
 	github.com/fortytw2/leaktest v1.3.0
 	github.com/frankban/quicktest v1.11.3 // indirect
+	github.com/jcmturner/gofork v1.0.0
 	github.com/jcmturner/gokrb5/v8 v8.4.2
 	github.com/klauspost/compress v1.12.2
 	github.com/kr/text v0.2.0 // indirect

--- a/gssapi_kerberos.go
+++ b/gssapi_kerberos.go
@@ -1,7 +1,6 @@
 package sarama
 
 import (
-	"encoding/asn1"
 	"encoding/binary"
 	"errors"
 	"fmt"
@@ -10,6 +9,7 @@ import (
 	"strings"
 	"time"
 
+	"github.com/jcmturner/gofork/encoding/asn1"
 	"github.com/jcmturner/gokrb5/v8/asn1tools"
 	"github.com/jcmturner/gokrb5/v8/gssapi"
 	"github.com/jcmturner/gokrb5/v8/iana/chksumtype"

